v2/tests/logger_int/main.c

```diff
@@ -0,0 +1,107 @@
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogWarning: a more complicated hello with int=%d.", 42);
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogWarning: the most complicated hello with int=%d and a really big string that fits =%s.", 42, really_big_string_fits);
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogWarning: the most complicated hello with int=%d and a really big string that does not fit=%s.", 42, really_big_string_doesnt_fit);
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogCritical: does PRI_BOOL work? %" PRI_BOOL "", MU_BOOL_VALUE(true) /*answer is yes, it works*/);
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogCritical: does PRI_BOOL work with amazingly out of bounds value? %" PRI_BOOL "", MU_BOOL_VALUE(42));
+    logger_log(LOG_LEVEL_WARNING, NULL, "LogCritical: does PRI_BOOL works with false? %" PRI_BOOL "", MU_BOOL_VALUE(false));
+
+    /*LogInfo*/
+    logger_log(LOG_LEVEL_INFO, NULL, "LogInfo: hello world!");
+    logger_log(LOG_LEVEL_INFO, NULL, "LogInfo: a more complicated hello with int=%d.", 42);
+    logger_log(LOG_LEVEL_INFO, NULL, "LogInfo: the most complicated hello with int=%d and a really big string that fits=%s.", 42, really_big_string_fits);
+    logger_log(LOG_LEVEL_INFO, NULL, "LogInfo: the most complicated hello with int=%d and a really big string that does not fit=%s.", 42, really_big_string_doesnt_fit);
+    logger_log(LOG_LEVEL_INFO, NULL, "LogCritical: does PRI_BOOL work? %" PRI_BOOL "", MU_BOOL_VALUE(true) /*answer is yes, it works*/);
+    logger_log(LOG_LEVEL_INFO, NULL, "LogCritical: does PRI_BOOL work with amazingly out of bounds value? %" PRI_BOOL "", MU_BOOL_VALUE(42));
+    logger_log(LOG_LEVEL_INFO, NULL, "LogCritical: does PRI_BOOL works with false? %" PRI_BOOL "", MU_BOOL_VALUE(false));
+
+    /*LogVerbose*/
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogVerbose: hello world!");
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogVerbose: a more complicated hello with int=%d.", 42);
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogVerbose: the most complicated hello with int=%d and a really big string that fits =%s.", 42, really_big_string_fits);
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogVerbose: the most complicated hello with int=%d and a really big string that does not fit=%s.", 42, really_big_string_doesnt_fit);
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogCritical: does PRI_BOOL work? %" PRI_BOOL "", MU_BOOL_VALUE(true) /*answer is yes, it works*/);
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogCritical: does PRI_BOOL work with amazingly out of bounds value? %" PRI_BOOL "", MU_BOOL_VALUE(42));
+    logger_log(LOG_LEVEL_VERBOSE, NULL, "LogCritical: does PRI_BOOL works with false? %" PRI_BOOL "", MU_BOOL_VALUE(false));
+
+    char* prop_value = malloc(2);
+    (void)strcpy(prop_value, "a");
+
+    // allocate a context dynamically
+    LOG_CONTEXT_HANDLE log_context;
+    LOG_CONTEXT_CREATE(log_context, NULL, LOG_CONTEXT_FIELD("property_name", "%s", MU_P_OR_NULL(prop_value)));
```

Comment:
>LOG_CONTEXT_CREATE [](http://example.com/codeflow?start=4&length=18)

can this fail? asking for a friend.

Reply:
The dynamic version can fail and the context is NULL in that case.